lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb

```diff
@@ -1186,13 +1187,16 @@ fc_extras
         cube.long_name = long_name
 
         if standard_name is not None:
-            if standard_name in iris.std_names.STD_NAMES:
-                cube.standard_name = standard_name
-            else:
-                if cube.long_name is not None:
-                    cube.attributes['invalid_standard_name'] = standard_name
+            try:
+                cube.standard_name = get_valid_standard_name(standard_name)
+            except ValueError as e:
+                if str(e) != '{!r} is not a valid standard_name'.format(standard_name):
+                    raise
                 else:
-                    cube.long_name = standard_name
+                    if cube.long_name is not None:
+                        cube.attributes['invalid_standard_name'] = standard_name
+                    else:
+                        cube.long_name = standard_name
```

Comment:
No? The diff that is being displayed is makign this a bit hard to read, but basically what was previously lines 1527-1538 
https://github.com/SciTools/iris/blob/ec46449b9705789268b77089a2161ff2c1d605c5/lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb#L1527-L1539
https://github.com/SciTools/iris/blob/b6775634eccbe29a45519c85a0da99a36dbc4886/lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb#L1536-L1547
But they have been indented out once more because they are in the else of the except block.

One change I have made is I have removed line 1539 in the old code (see the last line in the first snippet above) because this line seemed pointless? It was setting long_name to be None even though it already was None?

Reply:
I don't believe so?
What were lines 1192-1195 in the old code
https://github.com/SciTools/iris/blob/ec46449b9705789268b77089a2161ff2c1d605c5/lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb#L1192-L1195
Are now lines 1196-1199 in the new code
https://github.com/SciTools/iris/blob/b6775634eccbe29a45519c85a0da99a36dbc4886/lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb#L1196-L1199

so no change, just an extra indentation due to being in the `else` of the `if`statement in the `except` part